lib/components/viewers/stop-schedule-table.js

```diff
@@ -0,0 +1,142 @@
+import moment from 'moment'
+import coreUtils from '@opentripplanner/core-utils'
+import React, { Component, createRef } from 'react'
+import styled from 'styled-components'
+
+import Loading from '../narrative/loading'
+import {FETCH_STATUS} from '../../util/constants'
+import {
+  getFirstDepartureFromNow,
+  mergeAndSortStopTimes
+} from '../../util/viewer'
+
+const { formatSecondsAfterMidnight } = coreUtils.time
+
+// Styles for the schedule table and its contents.
+const StyledTable = styled.table`
+  border-spacing: collapse;
+  height: 100%;
+  width: 100%;
+  th {
+    background-color: #fff;
+    box-shadow: 0 1px 0px 0px #ccc;
+    font-size: 75%;
+    position: sticky;
+    top: 0px;
+  }
+  tr > * {
+    border-bottom: 1px solid #ccc;
+    padding: 2px 0 2px 10px;
+    vertical-align: top;
+  }
+  td:first-child, th:first-child {
+    padding-left: 0;
+  }
+`
+
+// Limit the width of the block cell if agency uses very long block IDs,
+// otherwise the table contents overflows the width of the side pane.
+// FIXME: Determine a good way to display long block ids.
+const BlockCell = styled.td`
+  max-width: 100px;
+  overflow-x: hidden;
+  text-overflow: ellipsis;
+`
+const RouteCell = styled.td`
+  font-weight: bold;
+`
+const TimeCell = styled.td`
+  font-weight: bold;
+  text-align: right;
+  white-space: nowrap;
+`
+
+/**
+ * Table showing scheduled departure times for the specified stop organized
+ * chronologically.
+ */
+class StopScheduleTable extends Component {
+  targetDepartureRef = createRef()
```

Comment:
Please add a comment about what this is used for.

Reply:
Updated in https://github.com/opentripplanner/otp-react-redux/pull/292/commits/bb270a7041700a19cc42eebf190a88fc04b0214c